rest-backend/src/main/java/com/sap/psr/vulas/backend/repo/DependencyRepository.java

```diff
@@ -22,8 +23,8 @@
 	
 	public static final ResultSetFilter<Dependency> FILTER = new ResultSetFilter<Dependency>();
 
-	@Query("SELECT dep FROM Dependency dep JOIN FETCH dep.lib l WHERE dep.id = :id")
-	List<Dependency> findById(@Param("id") Long id);
+//	@Query("SELECT dep FROM Dependency dep JOIN FETCH dep.lib l WHERE dep.id = :id")
+//	Optional<List<Dependency>> findById(@Param("id") ID id);
```

Comment:
Can this change come with a performance penalty?

Reply:
I think so but the only place i found where it is used is the endpoint retrieving vulndeps for *all* applications which is not used anywhere in the frontend nor in the client and was already having performance issues; thus would probably require rework to scale.